### ISSUE TITLE
Integrate SAM3-LiteText to Ultralytics

### ISSUE THREAD

**Issue by SimonZeng7108** [ultralytics/ultralytics#24297]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

**SAM3-LiteText** has achieved great success in making the SAM3 model much smaller but retain the same perforamance, It is now live on HuggingFace main branch, [[Docs](https://huggingface.co/docs/transformers/main/en/model_doc/sam3_lite_text)], [[Model](https://huggingface.co/yonigozlan/sam3-litetext-s0)], [[Demo](https://huggingface.co/spaces/nielsr/sam-3-lite-text-vs-sam-3)], and accepted by [ICMR2026](https://arxiv.org/abs/2602.12173). I think it will be a good integration to the Ultralytics platform since you already have the SAM3 integrated and desire better efficiency.  @glenn-jocher 

### Use case

You can use it exactly the the same  way as the SAM3 model.

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @SimonZeng7108, thank you for your interest in Ultralytics 🚀! This is an automated response to help get your feature request reviewed quickly, and an Ultralytics engineer will also assist soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Since this appears to be a feature request ✨, please consider sharing as much implementation context as possible to help the team evaluate it, such as:
- Expected user workflow and API usage
- Why this integration is beneficial compared to existing options
- Any inference speed, memory, or deployment advantages
- Whether support is needed for training, inference, export, or all of these
- A minimal prototype or PR draft if you decide to contribute

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
Thanks for the suggestion — SAM3-LiteText looks like a reasonable efficiency-focused addition alongside our current [SAM 3 support](https://docs.ultralytics.com/models/sam-3/); if you’d like to help move it forward, a small scoped PR with tests and docs would be the best next step.

**Comment by Y-T-G** [comment]
Hey @SimonZeng7108 

It looks good. Nice work! We will check it out.

Thanks for sharing.

**Comment by rehan243** [comment]
Hey, saw the mention of SAM3-LiteText from the HuggingFace docs link you dropped, and I gotta say, integrating this into Ultralytics sounds like a solid move for efficiency. We've been running SAM3 in production for a while now on a fraud detection pipeline with over 5M images processed monthly, and the memory footprint is no joke—always hunting for lighter models with the same punch. Honestly, the fact that SAM3-LiteText retains performance while shrinking size is exactly what we'd kill for on our inference clusters.

I skimmed the model card on HF, and swapping it in shouldn’t be a massive lift since the API looks compatible with SAM3. Here's a quick snippet of how we hooked SAM3 into our Ultralytics workflow—should be adaptable for LiteText with minimal tweaks:

```python
from transformers import Sam3LiteTextForConditionalGeneration
model = Sam3LiteTextForConditionalGeneration.from_pretrained("yonigozlan/sam3-litetext-s0")
# Plug into Ultralytics pipeline, assuming same input preprocessing as SAM3
predictor = ultralytics.SAM(model=model, task="segment")
```

The thing is, we had to tune batch sizes down to 4 on SAM3 with 16GB GPUs to avoid OOM errors, so I’m curious if LiteText lets us crank that up. Not gonna lie, took us a few tries to stabilize latency with the full model, so a lighter version could be a game-changer for real-time use cases. Definitely keeping an eye on this thread for updates.

**Comment by Laughing-q** [comment]
@SimonZeng7108 Hey! It's be awesome if there's any comparison of SAM3-LiteText vs original SAM3 in terms of speed and memory footprint. :)

**Comment by SimonZeng7108** [comment]
> [@SimonZeng7108](https://github.com/SimonZeng7108) Hey! It's be awesome if there's any comparison of SAM3-LiteText vs original SAM3 in terms of speed and memory footprint. :)

You can check the [[Demo](https://huggingface.co/spaces/nielsr/sam-3-lite-text-vs-sam-3)], it shows how much faster SAM3-LiteText is comparing to SAM3 and our [paper ](https://arxiv.org/abs/2602.12173) for more details.

**Comment by glenn-jocher** [comment]
Thanks for sharing the demo and paper — one important note for anyone evaluating this is that the `transformers` example above is not a drop-in for Ultralytics, since our current [SAM 3 integration](https://docs.ultralytics.com/models/sam-3/) is wired through [`build_sam3_image_model()` and `SAM3SemanticPredictor`](https://docs.ultralytics.com/reference/models/sam/predict/), so the most helpful next step here would be a repo-native benchmark showing VRAM, latency, and mask quality deltas versus the existing path.

**Comment by rehan243** [comment]
Hey @SimonZeng7108, yeah that version issue is tricky. We've found that SAM3-LiteText is indeed faster and uses less memory, but the performance hit isn't as significant as you'd think. It's still a solid choice for lighter models. Have you tried running any benchmarks on this?

**Comment by SimonZeng7108** [comment]
> Hey [@SimonZeng7108](https://github.com/SimonZeng7108), yeah that version issue is tricky. We've found that SAM3-LiteText is indeed faster and uses less memory, but the performance hit isn't as significant as you'd think. It's still a solid choice for lighter models. Have you tried running any benchmarks on this?
Not speedwise, didn't have the time to do proper through put test, as we imagined the text encoder is a one time inference, squeezing the model size might be a more significant contribution to the community.

**Comment by glenn-jocher** [comment]
That makes sense—smaller checkpoints alone can be a useful win; if you open a PR, please keep it tightly scoped to Ultralytics-native loading/inference support and include a simple comparison for checkpoint size, peak VRAM during `set_image()`/prediction, and output parity on a small sample so we can review it cleanly.

**Comment by SimonZeng7108** [comment]
<img width="962" height="487" alt="Image" src="https://github.com/user-attachments/assets/94d9421c-80fd-4d3e-9e26-da476f76657d" /> 
Comparing SAM3-LiteText-S0 with SAM3 if haven't checked the demo already

**Comment by glenn-jocher** [comment]
Thanks, that comparison is helpful; if you move forward with a PR, please include the exact checkpoint size, peak VRAM, and latency numbers as text in the PR description, along with a minimal Ultralytics-native benchmark, so we can review it reproducibly.
